jax/interpreters/xla.py

```diff
@@ -609,6 +610,13 @@ def _xla_callable(fun: lu.WrappedFun, device, backend, name, donated_invars, *ar
   log_priority = logging.WARNING if FLAGS.jax_log_compiles else logging.DEBUG
   logging.log(log_priority, "Compiling %s for args %s.", fun.__name__, abstract_args)
 
+  if nreps > 1:
+    print(f"The jitted function {fun.__name__} includes a pmap. Using "
+          "jit-of-pmap can lead to inefficient data movement, as the outer jit "
+          "does not preserve sharded data representations and instead collects "
+          "input and output arrays onto a single device. "
```

Comment:
Maybe link to https://github.com/google/jax/issues/2926? Mostly to signal that we'd like to fix this, and also so people can track when it's fixed.

Reply:
Done!